packages/xrpl/src/models/transactions/AMMDeposit.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable complexity -- required for validateAMMDeposit */
+import { ValidationError } from '../../errors'
+import { Amount, IssuedCurrencyAmount } from '../common'
+
+import {
+  BaseTransaction,
+  isAmount,
+  isIssuedCurrency,
+  validateBaseTransaction,
+} from './common'
+
+/**
+ * AMMDeposit is the deposit transaction used to add liquidity to the AMM instance pool,
+ * thus obtaining some share of the instance's pools in the form of LPToken.
+ *
+ * The following are the recommended valid combinations:
+ * - LPToken
+ * - Asset1In
+ * - Asset1In and Asset2In
+ * - Asset1In and LPToken
+ * - Asset1In and EPrice
+ */
+export interface AMMDeposit extends BaseTransaction {
+  TransactionType: 'AMMDeposit'
+
+  /**
+   * A hash that uniquely identifies the AMM instance. This field is required.
+   */
+  AMMID: string
+
+  /**
+   * Specifies the amount of shares of the AMM instance pools that the trader
+   * wants to redeem or trade in.
+   */
+  LPToken?: IssuedCurrencyAmount
+
+  /**
+   * Specifies one of the pool assets (XRP or token) of the AMM instance to
+   * deposit more of its value.
+   */
+  Asset1In?: Amount
+
+  /**
+   * Specifies the other pool asset of the AMM instance to deposit more of
+   * its value.
+   */
+  Asset2In?: Amount
+
+  /**
+   * Specifies the maximum effective-price that LPToken can be traded out.
+   */
+  EPrice?: number
```

Comment:
This should be a string or Amount object (Depending if the Asset1 being deposited is XRP or not) - According to section 2.3.1

Reply:
Ah, copy paste error. Good catch :)

Reply:
Ah, copy/paste typo. Good catch :)